scripts/build-metrics.py

```diff
@@ -0,0 +1,296 @@
+            "wall_time": get_epoch(log_lines[-1]) - get_epoch(log_lines[0]),
+        }
+
+        for line in log_lines:
+            start, end, epoch, object_path, _ = line.split()
+            start = int(start)
+            end = int(end)
+            duration = ms2sec(end - start)
+
+            # Don't track dependency times (refine check potentially?)
+            if not object_path.startswith("velox"):
+                continue
+
+            _, ext = splitext(object_path)
+            if ext in [".so", ".a"] or not ext:
+                totals["link_time"] += duration
+                suite = "linking"
+            elif ext == ".o":
+                totals["compile_time"] += duration
+                suite = "compiling"
+            else:
+                print(f"Unkown file type found: {object_path}")
+                print("Skipping...")
+                continue
+
+            time_result = BenchmarkResult(
+                run_reason="merge",
+                batch_id=batch_id,
+                stats={
+                    "data": [duration],
+                    "unit": "s",
+                    "iterations": 1,
+                },
+                tags={
+                    "name": object_path,
+                    "suite": suite,
+                    "source": f"{self.build_type}_build_metrics_time",
+                },
+                info={},
+                context={"benchmark_language": "C++"},
+            )
+            results.append(time_result)
+
+        totals["total_time"] = totals["link_time"] + totals["compile_time"]
+        for total_name, total in totals.items():
+            total_result = BenchmarkResult(
+                run_reason="merge",
+                batch_id=batch_id,
+                stats={
+                    "data": [total],
+                    "unit": "s",
+                    "iterations": 1,
+                },
+                tags={
+                    "name": total_name,
+                    "suite": "total",
+                    "source": f"{self.build_type}_build_metrics_time",
+                },
+                info={},
+                context={"benchmark_language": "C++"},
+            )
+            results.append(total_result)
+
+        return results
+
+
+# find velox -type f -name '*.o' -exec ls -l -BB {} \; | awk '{print $5, $9}' |  sed 's|CMakeFiles/.*dir/||g' > /tmp/object-size
+
+
+def upload(args):
+    print("Uploading Build Metrics")
+    pr_number = int(args.pr_number) if args.pr_number else None
+    run_reason = "pull request" if pr_number else "commit"
+    run_name = f"{run_reason}: {args.sha}"
+    sizes = BinarySizeAdapter(
+        command=["true"],
+        size_file=join(args.base_path, args.size_file),
+        build_type=args.build_type,
+        result_fields_override={
+            "run_id": args.run_id,
+            "run_name": run_name,
+            "run_reason": run_reason,
+            "github": {
+                # TODO change
```

Comment:
nit: what should be changed, can you add a more detailed comment.

Reply:
ah, ups ^^ (it had my fork url before)